action.yaml

```diff
@@ -394,16 +394,16 @@ runs:
           }
 
           # skip if already there
-          if (pkg_ref$ref %in% d$get_field("Config/Needs/DepsDev")) {
+          if (pkg_ref$ref %in% d$get_field("Config/Needs/DepsDev", ""))) {
             cli::cli_alert_info("Reference for package {.pkg {pkg}} already exists in the {.field Config/Needs/DepsDev} field. Skipping.")
             next()
           }
 
           # skip if already in `DepsBranch`
           # this is to avoid duplicates - `Deps Branch` takes precedence over `DepsDev`
           if (
-            (!identical(d$get_field("Config/Needs/DepsBranch"), "")) &&
-            (pkg %in% vapply(d$get_field("Config/Needs/DepsBranch"), function(x) pkgdepends::parse_pkg_ref(x)$package, character(1)))
+            (!identical(d$get_field("Config/Needs/DepsBranch", ""), "")) &&
+            (pkg %in% vapply(d$get_field("Config/Needs/DepsBranch", ""), function(x) pkgdepends::parse_pkg_ref(x)$package, character(1)))
```

Comment:
in addition to the above comment, `pkgdepends::parse_pkg_ref()` fails on empty string so the trick with `""` default value won't work

```
r$> pkgdepends::parse_pkg_ref("")
Error in `get_remote_types(refs)`:
! Cannot parse package: .
ℹ See `?pkgdepends::pkg_refs()` for supported package sources.
Type .Last.error to see the more details.
```

We probably need to introduce an explicit check for existance -> `d$has_field`. This is not a replacement for a check of empty value - we need to check this as well

Reply:
The "" field shouldn't be used as input on `parse_pkg_ref()` because there is a `desc_field_clear_if_empty()` . But I added a condition to skip parse_pkg_ref if it ever reach this point in the code.